src/main/java/org/jetbrains/bsp/bazel/server/bsp/services/ScalaBuildServerService.java

```diff
@@ -54,10 +63,55 @@ public ScalaBuildServerService(BazelData bazelData, BazelRunner bazelRunner) {
     return CompletableFuture.completedFuture(new ScalaTestClassesResult(new ArrayList<>()));
   }
 
-  public CompletableFuture<ScalaMainClassesResult> buildTargetScalaMainClasses(
+  public Either<ResponseError, ScalaMainClassesResult> buildTargetScalaMainClasses(
       ScalaMainClassesParams scalaMainClassesParams) {
-    System.out.printf("DWH: Got buildTargetScalaMainClasses: %s%n", scalaMainClassesParams);
-    // TODO(illicitonion): Populate
-    return CompletableFuture.completedFuture(new ScalaMainClassesResult(new ArrayList<>()));
+    TargetRulesResolver<ScalaMainClassesItem> targetRulesResolver =
```

Comment:
it doesn't have to be created on every method call - i think that it should be moved to the constructor